python_modules/dagster/dagster/_core/definitions/reconstruct.py

```diff
@@ -739,30 +740,37 @@ def repository_def_from_target_def(
         # reassign to handle both repository and pending repo case
         target = target.get_inner_repository()
 
+    repo_def: Optional[RepositoryDefinition] = None
     # special case - we can wrap a single job in a repository
     if isinstance(target, (JobDefinition, GraphDefinition)):
         # consider including job name in generated repo name
-        return RepositoryDefinition(
+        repo_def = RepositoryDefinition(
             name=get_ephemeral_repository_name(target.name),
             repository_data=CachingRepositoryData.from_list([target]),
         )
     elif isinstance(target, list) and all(
         isinstance(item, (AssetsDefinition, SourceAsset)) for item in target
     ):
-        return RepositoryDefinition(
+        repo_def = RepositoryDefinition(
             name=SINGLETON_REPOSITORY_NAME,
             repository_data=CachingRepositoryData.from_list(target),
         )
     elif isinstance(target, RepositoryDefinition):
-        return target
+        repo_def = target
     elif isinstance(target, PendingRepositoryDefinition):
         # must load repository from scratch
         if repository_load_data is None:
-            return target.compute_repository_definition()
+            repo_def = target.compute_repository_definition()
         # can use the cached data to more efficiently load data
-        return target.reconstruct_repository_definition(repository_load_data)
-    else:
-        return None
+        else:
+            repo_def = target.reconstruct_repository_definition(repository_load_data)
```

Comment:
would be nice to extract this out to a function and have a single variable assignment so we know there are no codepaths that miss an assignment

Reply:
Good call, updated